admin/src/main/scala/io/buoyant/admin/names/DelegateApiHandler.scala

```diff
@@ -126,6 +137,44 @@ object DelegateApiHandler {
         }
     }
 
+    def parseBound(bound: Bound): Name.Bound = Name.Bound(Var(Addr.toFinagle(bound.addr)), bound.id, bound.path)
+
+    def toDelegateTree(jdt: JsonDelegateTree): DelegateTree[Name.Bound] = jdt match {
+      case Empty(p, d) => DelegateTree.Empty(p, d.getOrElse(Dentry.nop))
+      case Fail(p, d) => DelegateTree.Fail(Path.read(p), d.getOrElse(Dentry.nop))
+      case Neg(p, d) => DelegateTree.Neg(Path.read(p), d.getOrElse(Dentry.nop))
+      case Exception(p, d, msg) => DelegateTree.Exception(p, d.getOrElse(Dentry.nop), new java.lang.Exception(msg))
+      case Delegate(p, d, delegate) => DelegateTree.Delegate(p, d.getOrElse(Dentry.nop), toDelegateTree(delegate))
+      case Leaf(p, d, bound) =>
+        DelegateTree.Leaf(p, d.getOrElse(Dentry.nop), parseBound(bound))
+      case Alt(p, d, alts) => DelegateTree.Alt(p, d.getOrElse(Dentry.nop), alts.map(toDelegateTree): _*)
+      case Union(p, d, weighteds) =>
+        val delegateTreeWeighteds = weighteds.map {
+          case Weighted(w, tree) =>
+            DelegateTree.Weighted(w, toDelegateTree(tree))
+        }
+        DelegateTree.Union(p, d.getOrElse(Dentry.nop), delegateTreeWeighteds: _*)
+      case Transformation(p, name, bound, tree) => DelegateTree.Transformation(p, name, parseBound(bound), toDelegateTree(tree))
```

Comment:
nit: would you mind breaking some of these long lines on `=>` so we keep things generally <100 chars?